lib/kafka_rest_client.js

```diff
@@ -76,8 +78,29 @@ function KafkaRestClient(options, callback) {
     self.scheduleTopicMapRefresh();
     self.statsd = (options.statsd && typeof options.statsd.increment === 'function') ? options.statsd : null;
     self.metricsPrefix = 'kafka_rest_proxy.rest-client.' + os.hostname() + '.';
+
+    if ('localAgentPort' in options) {
+        self.localAgentPort = options.localAgentPort;
+    } else {
+        self.localAgentPort = defaultLocalAgentPort;
+    }
+
+    if ('produceInterval' in options) {
+        self.produceInterval = options.produceInterval;
+    } else {
+        self.produceInterval = -1;
+    }
 }
 
+KafkaRestClient.prototype.getProduceInterval = function getProduceInterval(retry) {
+    var self = this;
+    if (self.produceInterval === -1) {
+        return 600 * Math.pow(5, retry + 1);
+    } else {
+        return self.produceInterval;
```

Comment:
guess produceInterval is for just test purpose. If so, better to add a comment, since in prod env exponential back-off is preferred.